--- vendor/device/qcom/lahaina/init.target.rc	2026-05-22 14:08:48.696226386 +0200
+++ vendor/device/qcom/lahaina/init.target.rc	2026-05-22 17:38:09.845172462 +0200
@@ -223,3 +223,23 @@
     user root
     disabled
     oneshot
+
+# Configure USB inferface as ADB via ConfigFS
+on property:sys.usb.config=adb && property:sys.usb.configfs=1
+    start adbd
+
+on property:sys.usb.ffs.ready=1 && property:sys.usb.config=adb && property:sys.usb.configfs=1
+    write /config/usb_gadget/g1/configs/b.1/strings/0x409/configuration "adb"
+    rm /config/usb_gadget/g1/configs/b.1/f1
+    rm /config/usb_gadget/g1/configs/b.1/f2
+    write /config/usb_gadget/g1/idVendor 0x05C6
+    write /config/usb_gadget/g1/idProduct 0x9012
+    symlink /config/usb_gadget/g1/functions/ffs.adb /config/usb_gadget/g1/configs/b.1/f1
+    write /config/usb_gadget/g1/UDC ${sys.usb.controller}
+    setprop sys.usb.state ${sys.usb.config}
+
+# USB/peripheral mode
+on property:sys.boot_completed=1
+    write /sys/bus/platform/devices/a600000.ssusb/mode peripheral
+    setprop persist.vendor.usb.config adb
+    setprop sys.usb.config adb
