--- chipcode/boot_images/edk2/BaseTools/Conf/tools_def.template	2026-05-24 08:49:55.019847961 +0200
+++ chipcode/boot_images/edk2/BaseTools/Conf/tools_def.template	2026-05-24 08:46:15.893094281 +0200
@@ -3249,8 +3249,8 @@
 *_CLANG100LINUX_AARCH64_VFRPP_FLAGS    = DEF(GCC_VFRPP_FLAGS) DEF(CLANG35_AARCH64_TARGET)
 *_CLANG100LINUX_AARCH64_ASLPP_FLAGS    = DEF(GCC_ASLPP_FLAGS) DEF(CLANG35_AARCH64_TARGET)
 
-DEBUG_CLANG100LINUX_AARCH64_CC_FLAGS   = DEF(CLANG100_AARCH64_CC_FLAGS) $(ARCHCC_FLAGS) $(PLATFORM_FLAGS) -mstrict-align -mcpu=cortex-a53
-RELEASE_CLANG100LINUX_AARCH64_CC_FLAGS = DEF(CLANG100_AARCH64_CC_FLAGS) $(ARCHCC_FLAGS) $(PLATFORM_FLAGS) -mstrict-align -mcpu=cortex-a53
+DEBUG_CLANG100LINUX_AARCH64_CC_FLAGS   = DEF(CLANG100_AARCH64_CC_FLAGS) $(ARCHCC_FLAGS) $(PLATFORM_FLAGS) -mstrict-align -mcpu=cortex-a53 -Wno-error=void-pointer-to-int-cast -Wno-error=pointer-to-int-cast -Wno-void-pointer-to-int-cast -Wno-pointer-to-int-cast
+RELEASE_CLANG100LINUX_AARCH64_CC_FLAGS = DEF(CLANG100_AARCH64_CC_FLAGS) $(ARCHCC_FLAGS) $(PLATFORM_FLAGS) -mstrict-align -mcpu=cortex-a53 -Wno-error=void-pointer-to-int-cast -Wno-error=pointer-to-int-cast -Wno-void-pointer-to-int-cast -Wno-pointer-to-int-cast
 
 
 ####################################################################################
